datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -5,30 +5,128 @@
  */
 package com.linkedin.datastream.server.assignment;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
-import org.apache.commons.lang.NotImplementedException;
+import com.google.common.annotations.VisibleForTesting;
 
 import com.linkedin.datastream.server.ClusterThroughputInfo;
+import com.linkedin.datastream.server.DatastreamGroupPartitionsMetadata;
 import com.linkedin.datastream.server.DatastreamTask;
+import com.linkedin.datastream.server.DatastreamTaskImpl;
+import com.linkedin.datastream.server.PartitionThroughputInfo;
 
 
 /**
  * Performs partition assignment based on partition throughput information
  */
 public class LoadBasedPartitionAssigner {
+
+  private static final Integer DEFAULT_KB_RATE = 5;
+  private static final Integer DEFAULT_MESSAGE_RATE = 5;
+
   /**
-   * Get partition assignment
-   * @param throughputInfo Per-partition throughput information
-   * @param assignedPartitions List of assigned partitions
-   * @param unassignedPartitions List of unassigned partitions
-   * @param taskCount Task count
-   * @return Partition assignment
+   * Performs partition assignment based on partition throughput information
+   * @param throughputInfo Per partition throughput information
+   * @param currentAssignment Current assignment
+   * @param unassignedPartitions Unassigned partitions
+   * @param partitionMetadata Partition metadata
+   * @return New assignment
    */
   public Map<String, Set<DatastreamTask>> assignPartitions(ClusterThroughputInfo throughputInfo,
-      List<String> assignedPartitions, List<String> unassignedPartitions, int taskCount) {
-    throw new NotImplementedException();
+      Map<String, Set<DatastreamTask>> currentAssignment, List<String> unassignedPartitions,
```

Comment:
Add validation that unassignedPartitions has no conflict with currentAssignment to avoid the partition assigned to 2 tasks. It will be difficult to debug in the production environment and we can easily rule out this possibility, if we have the validation in the code itself.

Reply:
I'll add it in `LoadBasedPartitionAssignmentStrategy`